src/languages/basic.js

```diff
@@ -198,7 +198,14 @@ export default function(hljs) {
       keyword: KEYWORDS
     },
     contains: [
-      hljs.QUOTE_STRING_MODE,
+      {
+        // Match strings that start with " and end with " or a line break
+        scope: 'string',
+        begin: /"/,
+        end: /"|$/,
+        illegal: /\n/,
```

Comment:
I'm not sure this has any effect at all since the `$` should match first (or does it match after)... are you seeing different behavior with/without `illegal`?

Reply:
I added the illegal keyword to ensure the string doesn't span multiple lines. But I can't see a different behavior without it. This can be removed to optimize the code.

Reply:
It can't span multiple lines because the `$` will match the end of the first line.